pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -9,9 +9,202 @@ import "google/protobuf/empty.proto";
 
 service ChainReader {
   rpc GetLatestValue (GetLatestValueRequest) returns (GetLatestValueReply) {}
+  rpc QueryKey (QueryKeyRequest) returns (QueryKeyReply) {}
+  rpc QueryKeys (QueryKeysRequest) returns (QueryKeysReply) {}
```

Comment:
do we need both the singular and pural to be exposed?

at least in the grpc layer, can't we expose the pural and, if needed elsewhere, adapt a signular use case to the pural?

that is, we could have golang API the for the singular and pural if needed, but only the plural here and shim the golang-grpc as needed.

Reply:
Most of the use cases are singular, so we would have to have an adapter everywhere.

Reply:
Plural can check incoming requests, if it's one it can call a different method internally.

Reply:
What i wanted to say is that most of the use cases are singular which means that almost every call would use an adapter which I think would look ugly.
On the other hand if you make an adapter over singular for plural it could have performance implications since you aren't making a potentially optimised plural call, but multiple singular calls through an adapter.